Expose option to disable CRC-32 computation during `torch.save`

mikaylagawarecki opened the pull request:
Option only works in open source, not internal

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137735

Differential Revision: [D64545647](https://our.internmc.facebook.com/intern/diff/D64545647)

mikaylagawarecki committed:
Expose option to disable CRC-32 computation

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Expose option to disable CRC-32 computation"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Expose option to disable CRC-32 computation during `torch.save`"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Expose option to disable CRC-32 computation during `torch.save`"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Expose option to disable CRC-32 computation during `torch.save`"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Expose option to disable CRC-32 computation during `torch.save`"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Expose option to disable CRC-32 computation during `torch.save`"




[ghstack-poisoned]

albanD commented on torch/serialization.py:
```diff
@@ -167,6 +169,34 @@ def set_default_load_endianness(endianness):
+    return _compute_crc32
+
+
+def set_default_crc32_options(compute_crc32: bool):
```
nit: why `set_default_crc32_options` vs `set_crc32_options` ? There is no way to override this flag from the end user of torch.save point of view.

mikaylagawarecki replied:
will use `set_crc32_options`

albanD commented on third_party/miniz-2.1.0/miniz.h:
```diff
@@ -1001,7 +1001,8 @@ typedef enum {
     MZ_ZIP_FLAG_WRITE_ALLOW_READING = 0x8000,
-    MZ_ZIP_FLAG_ASCII_FILENAME = 0x10000
+    MZ_ZIP_FLAG_ASCII_FILENAME = 0x10000,
+    MZ_ZIP_FLAG_DO_NOT_COMPUTE_CRC32 = 0x20000, /* don't compute the crc32 of file data that's being added. */
```
I forgot what's the story with these changes and if they will work internally?

mikaylagawarecki replied:
won't work internally as internal uses a different version of miniz

albanD reviewed (commented):
Only small nits, sounds pretty good in general!

mikaylagawarecki committed:
Update on "Expose option to disable CRC-32 computation during `torch.save`"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Expose option to disable CRC-32 computation during `torch.save`"

Option only works in open source, not internal




[ghstack-poisoned]

albanD reviewed (approved):
SGTM !

mikaylagawarecki commented:
@pytorchbot merge

clee2000 commented:
@pytorchbot revert -m "failing internally D64438525, probably needs gating" -c ghfirst

```
serialize/inline_container.cc:701:14: error: use of undeclared identifier 'MZ_ZIP_FLAG_DO_NOT_COMPUTE_CRC32'
    flags |= MZ_ZIP_FLAG_DO_NOT_COMPUTE_CRC32;
```

mikaylagawarecki committed:
Update on "Expose option to disable CRC-32 computation during `torch.save`"

Option only works in open source, not internal




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Expose option to disable CRC-32 computation during `torch.save`"

Option only works in open source, not internal


Differential Revision: [D64545647](https://our.internmc.facebook.com/intern/diff/D64545647)

[ghstack-poisoned]

mikaylagawarecki commented:
@mikaylagawarecki has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D64545647).

mikaylagawarecki committed:
Update on "Expose option to disable CRC-32 computation during `torch.save`"

Option only works in open source, not internal


Differential Revision: [D64545647](https://our.internmc.facebook.com/intern/diff/D64545647)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Expose option to disable CRC-32 computation during `torch.save`"

Option only works in open source, not internal


Differential Revision: [D64545647](https://our.internmc.facebook.com/intern/diff/D64545647)

[ghstack-poisoned]

wdvr commented:
@pytorchmergebot merge -f "already landed internally"

mikaylagawarecki commented:
@pytorchmergebot merge -f "already landed internally"

wdvr commented:
as discussed with @mikaylagawarecki and @izaitsevfb - this has been reimported as a different PR to circumvent internal export check failures. I will close this one unmerged if that one is merged in.
